backend/src/services/trading_service.py

```
@@ -175,57 +204,153 @@ async def execute_sell_order(self, token_address: str, amount_tokens: float, sli
+        retry_interval = 2.0
+        while (datetime.now().timestamp() - start_time) < timeout:
+            try:
+                response = await self.solana_client.get_signature_statuses([signature])
+                if response.value and response.value[0]:
+                    status = response.value[0]
+                    if status.err:
+                        logger.error(f"Transaction failed: {status.err}")
+                        return False
+                    if status.confirmation_status in ['confirmed', 'finalized']:
+                        return True
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)